components/script/dom/abstractworkerglobalscope.rs

```diff
@@ -110,19 +111,23 @@ pub fn run_worker_event_loop<T, WorkerMsg, Event>(
         + DomObject,
 {
     let scope = worker_scope.upcast::<WorkerGlobalScope>();
-    let devtools_receiver = scope.devtools_receiver();
     let task_queue = worker_scope.task_queue();
+
+    let never = crossbeam_channel::never();
+    let devtools_receiver = scope.devtools_receiver().unwrap_or(&never);
+
     let event = select! {
         recv(worker_scope.control_receiver()) -> msg => T::from_control_msg(msg.unwrap()),
         recv(task_queue.select()) -> msg => {
             task_queue.take_tasks(msg.unwrap());
             T::from_worker_msg(task_queue.recv().unwrap())
         },
-        recv(devtools_receiver.unwrap_or(&crossbeam_channel::never())) -> msg =>
-            T::from_devtools_msg(msg.unwrap()),
+        recv(devtools_receiver) -> msg => T::from_devtools_msg(msg.unwrap()),
+        recv(scope.timer_scheduler().wait_channel()) -> _ => T::from_timer_msg(),
```

Comment:
For the same reason as in the other comment, I think it would be better to do `worker_scope.handle_event(T::from_timer_msg(), can_gc)` here, instead of adding the event to `sequential`(which in this case should be left empty).

Reply:
On second thoughts, I think we should dispatch the timers that are ready regardless of which event woke up up the loop, like for the window loop.